tools/release/update_setup_requirements.py

```diff
@@ -8,45 +8,87 @@
-    if publish_on_py_pi:
-        return f"{_line}=={version}\n"
-    tag = f"{version}-{_line.split('-')[1]}"
-    tar_name = f"{_line}-{version}"
-    return f"{_line} @ https://github.com/Avaiga/taipy/releases/download/{tag}/{tar_name}.tar.gz\n"
+def usage() -> None:
+    packages = "> <".join(f"{p}_ver" for p in PACKAGES)
+    print(  # noqa: T201
+        f"Usage: {sys.argv[0]} <package> <{packages}> <deps> [<gh_path>]"
+    )
+    packages = ", ".join(f"'{p}'" for p in PACKAGES[:-1])
+    packages = f"{packages}, or '{PACKAGES[-1]}'"
+    print(f"   <package> must be one of {packages}.")  # noqa: T201
+    for p in PACKAGES:
+        print(f"   <{p}_ver>: minimal version of the taipy-{p} dependency.")  # noqa: T201
+    print("   <deps> must be 'Pypi' or 'GitHub', indicating where to find Taipy package dependencies.")  # noqa: T201
+    print("   <gh_path>: The path of GitHub repository (owner/repo), used if <deps> is 'GitHub'.")  # noqa: T201
+
 
+def __build_taipy_package_line(line: str, version: str, use_pypi: bool, gh_path: t.Optional[str]) -> str:
+    line = line.strip()
+    if use_pypi:
+        # Target dependency version should the latest compatible with 'version'
+        major, minor = version.split(".")[:2]
+        return f"{line} >={major}.{minor},<{major}.{int(minor) + 1}\n"
+    tag = f"{version}-{line.split('-')[1]}"
+    tar_name = f"{line}-{version}"
+    return f"{line} @ https://github.com/{gh_path}/releases/download/{tag}/{tar_name}.tar.gz\n"
 
-def update_setup_requirements(package: str, versions: Dict, publish_on_py_pi: bool) -> None:
-    _path = os.path.join(BASE_PATH, package, "setup.requirements.txt")
+
+def update_setup_requirements(
+    package: str, versions: dict[str, str], publish_on_py_pi: bool, gh_path: t.Optional[str]
+) -> None:
+    path = os.path.join(BASE_PATH, "taipy" if package == "taipy" else f"taipy-{package}", "setup.requirements.txt")
     lines = []
-    with open(_path, mode="r") as req:
+    with open(path, mode="r") as req:
         for line in req:
-            if v := versions.get(line.strip()):
-                line = __build_taipy_package_line(line, v, publish_on_py_pi)
+            if match := re.match(r"^taipy(:?\-\w+)?\s*", line, re.MULTILINE):
+                # Add subpackage version if not forced
+                if not line[match.end():] and (v := versions.get(line.strip())):
+                    if v.startswith("0.0"):
+                        raise ValueError(f"Missing version for dependency '{line.strip()}'.")
+                    line = __build_taipy_package_line(line, v, publish_on_py_pi, gh_path)
             lines.append(line)
 
-    with open(_path, "w") as file:
+    with open(path, "w") as file:
         file.writelines(lines)
+    if True:  # More logs
+        print(f"Generated setup.requirements.txt for package '{package}'")  # noqa: T201
+        for line in lines:
+            print(line.strip())  # noqa: T201
+        print("-" * 32)  # noqa: T201
```

Comment:
I guess you wanted to prepare the future to make this piece of code de-activable. I would not. It does not cost much to have it all the time and it can always be useful.

```suggestion
    # More logs
    print(f"Generated setup.requirements.txt for package '{package}'")  # noqa: T201
    for line in lines:
        print(line.strip())  # noqa: T201
    print("-" * 32)  # noqa: T201
```